From e6b05485efb9eaf4abed74e6fefef3a806a806c9 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 11 Oct 2023 18:08:18 +0200
Subject: [PATCH 0189/1695] clk: stm32mp25: add CLK_SET_RATE_UNGATE flag to
 Flexgen clocks

Flexgens clock need to be ungate on a set_rate ops.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ee4d41652766c0f9126f8a0457937ebf81ae1c79 clk: stm32mp25: add CLK_SET_RATE_UNGATE flag to Flexgen clocks]
Change-Id: Ia7a9a3e7cc129c4a3002d12be4be906c9b7e5e9c
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>x
---
 core/drivers/clk/clk-stm32mp25.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index 35bf2c54b..f48ef9649 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -2354,7 +2354,7 @@ struct clk _name = {\
 		.flex_id	= (_flex_id),\
 	},\
 	.name		= #_name,\
-	.flags		= (_flags),\
+	.flags		= (_flags | CLK_SET_RATE_UNGATE),\
 	.num_parents	= 15,\
 	.parents	=  XBAR_PARENTS,\
 }
-- 
2.47.2

